lib/maybe.test.js

```diff
@@ -0,0 +1,46 @@
+const load = require('@std/esm')(module)
+const test = require('ava')
+const maybe = load('./maybe.mjs').default
+const curry = load('./curry.mjs').default
+
+test('maybe.open', t => {
+  t.is(maybe(4).open('isEmpty', x => x), 4, 'should process the value')
+  t.is(maybe().open('isEmpty', x => x), 'isEmpty', 'should return empty option')
+})
+
+test('maybe.isNothing', t => {
```

Comment:
Is `NaN` nothing?

Reply:
From [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/NaN)...

> The global NaN property is a value representing Not-A-Number.

So it is technically a value. However it may be more useful to add a test such as `isNaN(contents)`

Reply:
Yeah. JS definitely considers it a value. I think there is a strong case for hiding that as a convenience and calling it a Nothing. Either way, I think a test for it is a good idea.

Reply:
I do not think we should bloat the codebase or api with NaN-checks if we decide to treat it as a value.